test/unit/test_input_validation.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+from fairlearn._input_validation import _validate_and_reformat_input
+
+
+def test_input_column_metadata_retained():
```

Comment:
Could we also have a test with @moprescu 's repro case? That would also have us test learners from somewhere other than scikit-learn

Reply:
@adrinjalali  mentioned that we should move perf tests out of this repo, and I don't necessarily disagree (but it's a little bit of work). That would mean the tempeh dependency isn't required anymore. Lightgbm used to be part of tempeh (but isn't anymore I think due to some breaking change). 
The question is whether we really want to add dependencies on specific packages like lightgbm. Then what is the threshold? Do we add pytorch? What about tensorflow?
I think you see where I'm going with this. I'd much rather have the perf tests run all these things, and have the functional parts being checked here. What do you think?

Reply:
It would be a dependency for the test, not fairlearn.

I do take your point about keeping testing focussed, but I'm not talking about a full battery of tests for each ML package. I just think that we should perhaps have a smoke test or two for each of them, in an effort to make sure that things like this don't happen.

Or if we're only going to check that as part of the perf tests, then those need to be inserted into the release.

Reply:
requirements.txt has all dependencies, though, not just fairlearn's.

I actually WANT to have a full battery of tests for each ML package :-)

Anyway, I'll add some smoke tests like you suggested.